src/Services/QueuedJobService.php

```diff
@@ -1533,6 +1579,271 @@ private function addJobHandlersToLogger(LoggerInterface $logger, QueuedJob $job,
         }
     }
 
+    /**
+     * Find any eligible jobs that qualify for an automated job retry and schedule their retry
+     *
+     * @throws ValidationException
```

Comment:
I don't see a `ValidationException` being thrown in this method?
If one is being thrown, probably this should include information about what condition it's thrown in.

Reply:
I think this comes from the `write()` call.

Reply:
```suggestion
     * @throws ValidationException if validation fails during call to write()
```
We don't actually have a consistent way to apply these currently - my preference would be to remove it and raise a card to discuss adding `@throws` annotations systematically. But it's not worth holding up the whole PR for so just clarifying when it's thrown will be fine.